pkg/util/intl/resolve.go

```diff
@@ -27,20 +27,18 @@ func Resolve(preferred []string, fallback string, supported []string) (int, lang
 	return -1, tag
 }
 
-func ResolveLocaleCode(resolved string, fallback string, supported []string) string {
-	var matcher = language.NewMatcher(SupportedLanguageTags(supported))
-	var locale language.Tag
-
-	locale, _, confidence := matcher.Match(language.MustParse(resolved))
-	if confidence == language.No {
-		locale, _ = language.Parse(fallback)
+// ResolveUnicodeCldr resolves language tag to Unicode CLDR language tag.
+func ResolveUnicodeCldr(lang language.Tag, fallback language.Tag) string {
+	var clrdTags []language.Tag
```

Comment:
The matcher is actually static, since the CldrLanguages are static. Let use init() to initialize a module-level private matcher.

language.NewMatcher is actually a very expensive operation. In 6cbdcbf8f1 we cached matcher instance because of this.